### PR TITLE
Add --print-deps-to-file option for easier parsable dependencies

### PR DESCRIPTION
**Pull request by hroncok**
Fixes https://github.com/fedora-python/tox-current-env/issues/13

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -14,10 +14,11 @@ The ``tox-current-env`` plugin adds two options:
    An attempt to run this with a Python version that doesn't match will fail
    (if ``tox`` is invoked from an Python 3.7 environment, any non 3.7 testenv will fail).
 
-``tox --print-deps-only``
+``tox --print-deps-only`` / ``--print-deps-to-file``
     Instead of running any ``commands``,
-    simply prints the declared dependencies in ``deps`` to the standard output.
+    simply prints the declared dependencies in ``deps`` to the standard output or specified file.
     This is useful for preparing the current environment for the above.
+    ``--print-deps-to-file`` will override the file, if it already exists.
```
Comment (encukou) [inline README.rst]:
```suggestion
    ``--print-deps-to-file`` will overwrite the file if it already exists.
```

File: src/tox_current_env/hooks.py [hunk 2]
```diff
@@ -20,11 +20,23 @@ def tox_addoption(parser):
         default=False,
         help="Don't run tests, only print the dependencies",
     )
+    parser.add_argument(
+        "--print-deps-to-file",
+        action="store",
+        dest="print_deps_path",
+        metavar="PATH",
+        default=None,
+        help="Like --print-deps-only, but to a file (overrides the file)",
```
Comment (encukou) [inline src/tox_current_env/hooks.py]:
```suggestion
        help="Like --print-deps-only, but to a file. Overwrites the file if it exists.",
```

File: src/tox_current_env/hooks.py [hunk 3]
```diff
@@ -20,11 +20,23 @@ def tox_addoption(parser):
...
+    )
 
 
 @tox.hookimpl
 def tox_configure(config):
     """Stores options in the config. Makes all commands external and skips sdist"""
+    if config.option.print_deps_path is not None:
+        config.option.print_deps_only = True
+        with open(config.option.print_deps_path, "w", encoding="utf-8") as f:
+            f.write("")
```
Comment (encukou) [inline src/tox_current_env/hooks.py]: Consider [os.truncate](https://docs.python.org/3/library/os.html#os.truncate)(path, 0) instead.

File: src/tox_current_env/hooks.py [hunk 4]
```diff
@@ -136,7 +148,10 @@ def tox_runtest(venv, redirect):
     """If --print-deps-only, prints deps instead of running tests"""
     config = venv.envconfig.config
     unsupported_raise(config, venv)
+    if config.option.print_deps_path is not None:
+        with open(config.option.print_deps_path, "a", encoding="utf-8") as f:
+            print(*venv.get_resolved_dependencies(), sep="\n", file=f)
+        return True
```
Comment (encukou) [inline src/tox_current_env/hooks.py]: The `return True` means `--print-deps-only` won't have an effect with `--print-deps-to-file`.
Should these options be mutually exclusive?

**Comment by encukou** [comment]
Looks like it works as intended!
Consider removing the `--print-deps-only` flag entirely. For humans, the output is quite lost in the other messages; for computers it's mixed with the status output.

File: src/tox_current_env/hooks.py (hunk 4 above)
Comment (hroncok) [reply src/tox_current_env/hooks.py]: They should.

File: src/tox_current_env/hooks.py (hunk 3 above)
Comment (hroncok) [reply src/tox_current_env/hooks.py]: actually the previous option also ensure we fail early if the file is not writable and I consider that good. with os.truncate the code gets more complicated:

 - if the file exists, we truncate it
 - if the file does not exist, we need to check if we can create it

**Comment by hroncok** [comment]
> Consider removing the `--print-deps-only` flag entirely.

Will consider that for a subsequent PR.

File: src/tox_current_env/hooks.py (hunk 4 above)
Comment (hroncok) [reply src/tox_current_env/hooks.py]: Done. Please re-review (added in extra commit, so it should be easier).

**Pull request merged**